File: Algorithm/557_ReverseWordsInAString3/main.cpp
#include <bits/stdc++.h>
using namespace std;
string reverseWords(string s)
{
    string output = "";
    string tempString = "";
    for (int i = s.size() - 1; i >= 0; i--)
    {
        if (s[i] != ' ')
        {
            tempString += s[i];
        }
        else
        {
            output = tempString + (output != "" ? " " : "") + output;
            tempString = "";
        }
    }
    if (tempString != "")
    {
        output = tempString + (output != "" ? " " : "") + output;
    }
    return output;
}
string reverseWords2(string s)
{
    for (int i = 0; i < s.size(); i++)
    {
        int start = i;
        int end = i;
        while (end < s.size() && s[end] != ' ')
            end++;
        i = end;
        end--;
        while (start < end)
        {
            swap(s[start], s[end]);
            start++;
            end--;
        }
    }
    return s;
}
int main()
{
    string a = "Let's take LeetCode contest";
    string output = reverseWords2(a);

    cout << output << endl;
    return 0;
}

File: Algorithm/557_ReverseWordsInAString3/main.py
def reverseWords(self, s: str) -> str:
    i = 0
    t = list(s)
    while i < len(t):
        start = i
        end = i
        while end < len(s) and s[end] != " ":
            end += 1
        i = end
        end -= 1
        while start < end:
            temp = s[start]
            t[start] = s[end]
            t[end] = s[start]
            start += 1
            end -= 1
        i += 1
    return ''.join(t)


a = "Let's take LeetCode contest"
ans = reverseWords(0, a)
print(ans)
